server/src/test/java/io/druid/server/coordination/ZkCoordinatorTest.java

```diff
@@ -191,6 +257,114 @@ public void tearDown() throws Exception
     tearDownServerAndCurator();
   }
 
+  /**
+   * Steps:
+   * 1. removeSegment() schedules a delete runnable that deletes segment files after 1 second,
+   * 2. addSegment() succesfully loads the segment and annouces it
+   * 3. scheduled delete task executes and realizes it should not delete the segment files.
+   */
+  @Test
+  public void testSegmentLoading1() throws Exception
+  {
+    zkCoordinator.start();
+
+    final DataSegment segment = makeSegment("test", "1", new Interval("P1d/2011-04-01"));
+
+    zkCoordinator.removeSegment(segment, new DataSegmentChangeCallback()
+    {
+      @Override
+      public void execute()
+      {
+        // do nothing
+      }
+    });
+
+    Assert.assertFalse(segmentsAnnouncedByMe.contains(segment));
+
+    zkCoordinator.addSegment(segment, new DataSegmentChangeCallback()
+    {
+      @Override
+      public void execute()
+      {
+        // do nothing
+      }
+    });
+
+    /*
+       make sure the scheduled runnable that "deletes" segment files has been executed.
+       Because another addSegment() call is executed, which removes the segment from segmentsToDelete field in
+       ZkCoordinator, the scheduled runnable will not actually delete segment files.
+     */
+    for (Runnable runnable : scheduledRunnable) {
+      runnable.run();
+    }
+
+    Assert.assertTrue(segmentsAnnouncedByMe.contains(segment));
+    Assert.assertFalse("segment files shouldn't be deleted", segmentLoader.getSegmentsInTrash().contains(segment));
+
+    zkCoordinator.stop();
+  }
+
+  /**
+   * Steps:
+   * 1. addSegment() succesfully loads the segment and annouces it
+   * 2. removeSegment() unannounces the segment and schedules a delete runnable that deletes segment files after 1 second
```

Comment:
"after 1 second" is not true anymore and shoud be removed from the comment... right?